src/batch/src/executor/postgres_query.rs

```diff
@@ -0,0 +1,192 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use futures_async_stream::try_stream;
+use futures_util::stream::StreamExt;
+use risingwave_common::catalog::{Field, Schema};
+use risingwave_common::row::OwnedRow;
+use risingwave_common::types::{DataType, Datum, Decimal, ScalarImpl};
+use risingwave_common::util::chunk_coalesce::DataChunkBuilder;
+use risingwave_pb::batch_plan::plan_node::NodeBody;
+use thiserror_ext::AsReport;
+use tokio_postgres;
+
+use crate::error::BatchError;
+use crate::executor::{BoxedExecutor, BoxedExecutorBuilder, DataChunk, Executor, ExecutorBuilder};
+use crate::task::BatchTaskContext;
+
+/// S3 file scan executor. Currently only support parquet file format.
+pub struct PostgresQueryExecutor {
+    schema: Schema,
+    host: String,
+    port: String,
+    username: String,
+    password: String,
+    database: String,
+    query: String,
+    identity: String,
+}
+
+impl Executor for PostgresQueryExecutor {
+    fn schema(&self) -> &risingwave_common::catalog::Schema {
+        &self.schema
+    }
+
+    fn identity(&self) -> &str {
+        &self.identity
+    }
+
+    fn execute(self: Box<Self>) -> super::BoxedDataChunkStream {
+        self.do_execute().boxed()
+    }
+}
+
+pub fn postgres_row_to_owned_row(
+    row: tokio_postgres::Row,
+    schema: &Schema,
+) -> Result<OwnedRow, BatchError> {
+    let mut datums = vec![];
+    for i in 0..schema.fields.len() {
+        let rw_field = &schema.fields[i];
+        let name = rw_field.name.as_str();
+        let datum = postgres_cell_to_scalar_impl(&row, &rw_field.data_type, i, name)?;
+        datums.push(datum);
+    }
+    Ok(OwnedRow::new(datums))
+}
+
+// TODO(kwannoel): Support more types, see postgres connector's ScalarAdapter.
+fn postgres_cell_to_scalar_impl(
+    row: &tokio_postgres::Row,
+    data_type: &DataType,
+    i: usize,
+    name: &str,
+) -> Result<Datum, BatchError> {
+    // We observe several incompatibility issue in Debezium's Postgres connector. We summarize them here:
+    // Issue #1. The null of enum list is not supported in Debezium. An enum list contains `NULL` will fallback to `NULL`.
+    // Issue #2. In our parser, when there's inf, -inf, nan or invalid item in a list, the whole list will fallback null.
+    let datum = match data_type {
+        DataType::Boolean
+        | DataType::Int16
+        | DataType::Int32
+        | DataType::Int64
+        | DataType::Float32
+        | DataType::Float64
+        | DataType::Date
+        | DataType::Time
+        | DataType::Timestamp
+        | DataType::Timestamptz
+        | DataType::Jsonb
+        | DataType::Interval
+        | DataType::Bytea => {
+            // ScalarAdapter is also fine. But ScalarImpl is more efficient
+            row.try_get::<_, Option<ScalarImpl>>(i)?
```

Comment:
It seems `Varchar` is not here, any reason?

Reply:
I followed the implementation in `connector::parser::postgres`. But under the hood looks like it just converts `Varchar` in pg to `Varchar` in RW. Let me add it.